test/integration/targets/iam_user/tasks/main.yml

```diff
@@ -0,0 +1,119 @@
+---
+- name: set up aws connection info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+
+  - name: ensure ansible user exists
```

Comment:
It'd be good to add some assertions here, additionally you can cross-check that the user_id returned from iam_user matches the user_id returned by iam_user_info (just in case the search was doing something screwy)

Reply:
https://github.com/ansible/ansible/pull/23382/commits/518e1f4cb5c43ce481747a143ebd103d01796cec has updates so the values returned can be validated against the user created directly.